app/assets/javascript/components/editor/editor.js

```diff
@@ -0,0 +1,145 @@
+import { Controller } from '@hotwired/stimulus';
+import DOMPurify from 'dompurify';
+
+const EditorController = class extends Controller {
+  static targets = ['editor', 'toolbar', 'formInput'];
+
+  static TOOLBAR_ACTIONS = {
+    ul: 'insertUnorderedList',
+  };
+
+  static ALLOWED_TAGS = ['p', 'ul', 'li', '#text'];
+
+  connect() {
+    this.enabled = true;
+
+    // this will disable IE11 but make sure enabled for JS tests
+    if ((typeof process !== 'undefined' && process.env.NODE_ENV !== 'test') || typeof window.ClipboardEvent === 'undefined') {
```

Comment:
Instead of hiding the toolbar after detecting that the browser cannot do something which relies on JS, I would progressively enhance by adding the toolbar and listening to events in browsers that _do_ support the features.

That way it works without JS and browsers without the right features (perhaps like IE11)